### ISSUE TITLE
Inconsistent `COMMAND`: COMMAND INFO DONT BEHAVE SAME WHEN USED ON SELF (ALSO NO RECURSION ON SUB COMMANDS WHEN USED INFO)(eg: COMMAND INFO COMMAND )

### ISSUE THREAD

**Issue by sa-k-shore** [DiceDB/dice#758]
## Steps to reproduce

COMMAND INFO COMMAND

## Expected output
redis behaviour : 
<img width="392" alt="Screenshot 2024-09-28 at 1 37 44 AM" src="https://github.com/user-attachments/assets/8a0566f0-e002-46b6-b477-0a72541c8cd9">

## Observed output

The observed output when the above set of commands when run on DiceDB
<img width="264" alt="Screenshot 2024-09-28 at 1 39 36 AM" src="https://github.com/user-attachments/assets/10159edd-cad6-4e30-b1d0-27f4b9e9f39f">

## Expectations for resolution

This issue will be considered resolved when the following things are done

1. register them based on commands not on command name or change in cmdMeta.Name for command .
2. make sure to recurse subcommands in case of command info .
3. addition of relevant test case to ensure we catch the regression.

**Comment by sa-k-shore** [comment]
i am working on command docs which will face same issue so i would like to get this assigned so i will do it in correct way for docs and correct it for info also , thanks

**Comment by lucifercr07** [comment]
@sa-k-shore assigned, thanks for contributing.

**Comment by sa-k-shore** [comment]
i have raised a Merge Request  #993   with changes for #641 . please review, thanks.

**Comment by arpitbbhayani** [comment]
Hello @sa-k-shore,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.

**Comment by sa-k-shore** [comment]
hi @arpitbbhayani we have merged this issue and websocket test cases i missed and i am picking it as another task
